### PR TITLE
839 disclaimer pop up

### LINKED ISSUE #839: Disclaimer pop up windows.
We need to make sure that we carry these disclaimer popped up windows over to the new system.

Image

We have one for the assistant.  And one for business requests data

These should show up  once every 30 days.    And not be shown.Every time they visit.

### PR DESCRIPTION
**Pull request by jeneerthan**

### CODE REVIEW

File: app/frontend/src/playground/components/PlaygroundDisclaimerDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+      <DialogActions
+        sx={{
+          pb: 3,
+          px: 3,
+          gap: 1,
+          flexWrap: "wrap",
+          flexDirection: { xs: "column-reverse", sm: "row" },
+          alignItems: { xs: "stretch", sm: "center" },
+          "& .MuiButton-root": {
+            minHeight: 44,
```
Comment (Cody-SSC) [inline app/frontend/src/playground/components/PlaygroundDisclaimerDialog.tsx:110]: Blocker:
WCAG 2.4.3 / 1.3.2 — Focus order conflicts with visual order on mobile

File: PlaygroundDisclaimerDialog.tsx, line 106

flexDirection: { xs: "column-reverse", sm: "row" } makes the Accept button first in the DOM (and therefore first in tab order) but renders it visually last on small screens. WCAG 1.3.2 (Meaningful Sequence) and 2.4.3 (Focus Order) both require that keyboard and screen-reader order matches the visual presentation.

Fix: 
Change line 106 from column-reverse to column, then swap the button order in lines 113–127 so the language toggle comes first in the DOM (top on mobile) and Accept comes second (bottom on mobile). This way visual order and DOM/tab order are identical on all screen sizes.

File: app/frontend/src/playground/components/PlaygroundDisclaimerDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      <DialogTitle id="playground-disclaimer-title">{t(contentKeys.title)}</DialogTitle>
+      <DialogContent>
+        <Typography id={descriptionId} component="p" sx={{ mb: 3 }}>
+          {t(contentKeys.body)}
+        </Typography>
+        <Typography component="p" sx={{ fontWeight: 700 }}>
+          {t(contentKeys.strong)}
+        </Typography>
+      </DialogContent>
+      <DialogActions
```
Comment (Cody-SSC) [inline app/frontend/src/playground/components/PlaygroundDisclaimerDialog.tsx:101]: Blocker:
WCAG 1.3.1 — Semantic emphasis missing on critical warning text

File: PlaygroundDisclaimerDialog.tsx, lines 97–99

<Typography component="p" sx={{ fontWeight: 700 }}> applies bold visually only. Screen readers do not announce fontWeight styling. The text "DO NOT SEND ANY PROTECTED INFORMATION IN THE CHATBOT." is a critical security notice and must carry semantic emphasis via a <strong> element (WCAG 1.3.1 — Info and Relationships). Wrap {t(contentKeys.strong)} in <strong> inside the Typography.

File: app/frontend/src/playground/components/PlaygroundDisclaimerDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+  if (!currentDisclaimer) {
+    return null;
+  }
+
+  const contentKeys = CONTENT_KEYS[currentDisclaimer];
+  const descriptionId = "playground-disclaimer-description";
+
+  return (
+    <Dialog
```
Comment (Cody-SSC) [inline app/frontend/src/playground/components/PlaygroundDisclaimerDialog.tsx:82]: Minor:
Explicit intent for a mandatory/blocking dialog

File: PlaygroundDisclaimerDialog.tsx, lines 80–87

The <Dialog> has no onClose handler, so Esc key and backdrop clicks already do nothing — but this is implicit. MUI's Dialog is designed with onClose in mind and the current silent-ignore behaviour is fragile. Add disableEscapeKeyDown and an explicit onClose that does nothing (with a comment explaining it is intentionally mandatory) to make the blocking intent clear and prevent accidental dismissal if an onClose is ever added later.

File: app/frontend/src/playground/components/PlaygroundDisclaimerDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+      open={shouldShow}
```
Comment (Cody-SSC) [inline app/frontend/src/playground/components/PlaygroundDisclaimerDialog.tsx]: Minor:
shouldShow is always true at its use site

File: PlaygroundDisclaimerDialog.tsx, line 52 and line 81

shouldShow is assigned currentDisclaimer !== null on line 52, but the early return on lines 74–76 (if (!currentDisclaimer) { return null; }) already handles the null case before the JSX is ever reached. By line 81, shouldShow can only ever be true. The variable name implies a conditional that has already been resolved and is therefore misleading. Simplify open={shouldShow} to just open or open={true}.

File: app/frontend/src/playground/utils/disclaimerState.ts [hunk 5]
```diff
@@ -0,0 +1,89 @@
+export type PlaygroundDisclaimerKey = "assistant" | "br";
+
+export interface PlaygroundDisclaimerState {
```
Comment (Cody-SSC) [inline app/frontend/src/playground/utils/disclaimerState.ts:3]: Minor:
Constants duplicated between e2e tests and app source

Files: app/frontend/e2e/disclaimer.spec.ts lines 3–4 and app/frontend/e2e/fixtures/PlaygroundPage.ts line 1

PLAYGROUND_DISCLAIMER_STORAGE_KEY and THIRTY_DAYS_MS are both already exported from disclaimerState.ts (lines 8–9). Both e2e files redefine them as literals. If the storage key is ever bumped to v2, the e2e tests will silently seed the wrong key and produce false positives. Import from the source if the Playwright config allows it, or at minimum add a comment on each copy pointing to disclaimerState.ts so reviewers know they must be kept in sync.

**Review by Cody-SSC — COMMENTED** [review]
What's well done:

fullScreen={isSmallScreen} and minHeight: 44 touch targets (line 109) — correct mobile pattern meeting WCAG 2.5.5.

aria-labelledby="playground-disclaimer-title" and aria-describedby={descriptionId} wired to real element IDs (lines 85–86) — correct WCAG 4.1.2.

parseState with isFiniteTimestamp guard and try/catch in disclaimerState.ts (lines 13–35) — safe localStorage parsing with no XSS surface.

acceptPlaygroundDisclaimer returns new state immutably — correct React state pattern.
addInitScript for localStorage seeding in PlaygroundPage.ts (lines 24–50) — correct Playwright approach, avoids race conditions with page load.

Unit tests in disclaimerState.test.ts cover expiry boundary conditions exactly (29d 23h 59m 59s vs exactly 30d).

v1 suffix on the storage key (disclaimerState.ts line 8) — good for future migrations.

**Review by Cody-SSC — APPROVED** [review]
